### ISSUE TITLE
Deploy OKE autoscaler

### ISSUE THREAD

**Issue by hyder** [oracle-terraform-modules/terraform-oci-oke#579]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

1. Deploy OKE autoscaler following the guidelines: https://docs.oracle.com/en-us/iaas/Content/ContEng/Tasks/contengusingclusterautoscaler.htm
2. Investigate an easier way to deploy
3. Add tolerations
4. Handle multiple nodepools: some node pools can be marked for autoscaling, others not.
5. Handle scenarios when nodepools are destroyed during an upgrade

### Potential Terraform Configuration

```hcl
# Copy-paste any Terraform configurations for how the requested feature may be used. 
```

**Comment by codex70** [comment]
I've got a couple of comments:

Firstly, seeing this error when deploying:
```
│ Error: remote-exec provisioner error
│
│   with module.oke.module.extensions.null_resource.deploy_cluster_autoscaler[0],
│   on .terraform/modules/oke/modules/extensions/autoscaler.tf line 65, in resource "null_resource" "deploy_cluster_autoscaler":
│   65:   provisioner "remote-exec" {
│
│ error executing "/tmp/terraform_1576243720.sh": Process exited with status 1
╵
```
If I login to the server and run the script, or re-run terraform it works as expected, so I suspect it may need a sleep or wait of some kind.

Second point, maybe put a link to the documentation about how to create tags (https://docs.oracle.com/en-us/iaas/Content/Tagging/Tasks/Create_tag_key_definition.htm#Create_tag_key_definition), I wasn't even aware of these tags before I tried to install and wasn't sure what was needed. Other option (which would be even better) is to code something like the following to create the tag/namespace:
```
resource "oci_identity_tag_namespace" "oke_tag_namespace" {
  #Required
  compartment_id = var.compartment_id
  description = var.oke_tag_namespace_description
  name = var.oke_tag_namespace_name
  count = var.deploy_tags ? 1:0
}

resource "oci_identity_tag" "pool_tag" {
  #Required
  description = var.pool_tag_description
  name = var.pool_tag_name
  tag_namespace_id = oci_identity_tag_namespace.oke_tag_namespace[0].id
  validator {
    #Required
    validator_type = var.pool_tag_validator_type
    values = var.pool_tag_validator_values
  }
  count = var.deploy_tags ? 1:0
}

variable "deploy_tags" {
  default = false
  description = "Deploy the OKE tag, only necessary for the first cluster (will fail if installing a second time)"
  type = bool
}

variable "oke_tag_namespace_name" {
  default = "oke"
  description = "Kubernetes Cluster Autoscaler tag namespace"
  type = string
}

variable "oke_tag_namespace_description" {
  default = "OKE Kubernetes Cluster Autoscaler tag namespace"
  description = "Kubernetes Cluster Autoscaler tag namespace description "
  type = string
}

variable "pool_tag_name" {
  default = "pool"
  description = "Kubernetes Cluster Autoscaler pool tag"
  type = string
}

variable "pool_tag_description" {
  default = "Tag key to indicate node pool purpose"
  description = "Kubernetes Cluster Autoscaler pool tag description"
  type = string
}

variable "pool_tag_validator_type" {
  default = "ENUM"
  description = "Specifies the type of validation: a static value"
  type = string
}

variable "pool_tag_validator_values" {
  default = ["autoscaler"]
  description = "Allowed values"
  type = list(string)
}

```
Obvious issue you have is setting deploy_tags. First install needs this set to true, any further installs set to false.

**Comment by hyder** [comment]
Autoscaler issue resolved in #576
